apps/console/src/features/roles/components/edit-role/edit-role-permission.tsx

```diff
@@ -326,15 +326,17 @@ export const UpdatedRolePermissionDetails: FunctionComponent<RolePermissionDetai
 
                     response.map((apiResource: APIResourceInterface) => {
                         const selectedAPIResource: AuthorizedAPIListItemInterface =
-                            authorizedAPIListForApplication.find(
+                            authorizedAPIListForApplication?.find(
                                 (api: AuthorizedAPIListItemInterface) => api.id === apiResource.id
                             );
 
-                        selectedAPIResourcesList.push({
-                            id: selectedAPIResource.id,
-                            name: selectedAPIResource.displayName,
-                            scopes: selectedAPIResource.authorizedScopes
-                        });
+                        if (selectedAPIResource) {
+                            selectedAPIResourcesList.push({
+                                id: selectedAPIResource.id,
```

Comment:
Let's add null checks. Check the other places as well.

```suggestion
                                id: selectedAPIResource?.id,
```

Reply:
If the value of `selectedAPIResource` is undefined, the if condition resolves to false. Therefore, null checks inside the if condition are not needed.